packages/lmi/tests/test_cost_tracking.py

```diff
@@ -369,3 +369,673 @@ async def track_callback(response):  # noqa: RUF029
+            )
+
+            # Long system prompt to enable caching
+            long_text = (
+                "You are a streaming response assistant. " * 250
+            )  # ~1000+ tokens
+
+            messages = [
+                Message(role="user", content="Count to 5."),
+            ]
+
+            system_with_cache = [
+                {
+                    "type": "text",
+                    "text": long_text,
+                    "cache_control": {"type": "ephemeral"},
+                }
+            ]
+
+            # First streaming call: Creates cache
+            chunks1: list[str] = []
+            result1 = await model.call_single(
+                messages, system=system_with_cache, callbacks=[chunks1.append]
+            )
+
+            # Second streaming call: Reads from cache
+            chunks2: list[str] = []
+            result2 = await model.call_single(
+                messages, system=system_with_cache, callbacks=[chunks2.append]
+            )
+
+            # Assertions: Verify cached tokens extracted in streaming mode
+            assert (
+                result1.cache_creation_tokens is not None
+                and result1.cache_creation_tokens > 0
+            ) or (
+                result1.cache_read_tokens is not None and result1.cache_read_tokens > 0
+            ), "Expected cache activity on first streaming call"
+            assert (
+                result2.cache_creation_tokens is not None
+                and result2.cache_creation_tokens > 0
+            ) or (
+                result2.cache_read_tokens is not None and result2.cache_read_tokens > 0
+            ), "Expected cache activity on second streaming call"
+
+            # Costs should be calculated
+            assert result1.cost_usd is not None
+            assert result1.cost_usd > 0
+            assert result2.cost_usd is not None
+            assert result2.cost_usd > 0
+
+            # CRITICAL: Verify cache discount in streaming mode
+            # If call 2 reads from cache while call 1 created it, verify discount
+            if (
+                result2.cache_read_tokens is not None
+                and result2.cache_read_tokens > 0
+                and result1.cache_creation_tokens is not None
+                and result1.cache_creation_tokens > 0
+            ):
+                assert result2.cost_usd < result1.cost_usd, (
+                    f"Streaming: cache read cost ({result2.cost_usd:.6f}) should be < "
+                    f"cache creation cost ({result1.cost_usd:.6f}). "
+                    "Cache discount not being applied in streaming!"
+                )
+
+            # Verify streaming actually happened (received chunks)
+            assert chunks1, "Expected streaming chunks on first call"
+            assert chunks2, "Expected streaming chunks on second call"
+
+    @pytest.mark.asyncio
+    @pytest.mark.vcr
+    async def test_openai_cache_with_streaming(self):
```

Comment:
Can we somehow combine this test with the Anthropic one, and use `pytest.mark.parameterize`?